src/Microsoft.TemplateEngine.Edge/Template/TemplateMatchInfoEqualityComparer.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Edge.Template
+{
+    public class TemplateMatchInfoEqualityComparer : IEqualityComparer<ITemplateMatchInfo>
+    {
+        public static IEqualityComparer<ITemplateMatchInfo> Default { get; } = new TemplateMatchInfoEqualityComparer();
+
+        public bool Equals(ITemplateMatchInfo x, ITemplateMatchInfo y)
+        {
+            return ReferenceEquals(x.Info, y.Info) || (x != null && y != null && x.Info != null && y.Info != null && string.Equals(x.Info.Identity, y.Info.Identity, StringComparison.Ordinal));
```

Comment:
if x or y were null, this reference equals would have thrown before making it to the null check. Null propagation may make things cleaner here